lib/Distribution/Helper/Discover.hs

```diff
@@ -54,6 +54,10 @@ findProjects dir = execWriterT $ do
   let stackYaml = dir </> "stack.yaml"
   whenM (liftIO $ doesFileExist stackYaml) $
     tell [Ex $ ProjLocStackYaml stackYaml]
+
+  join $ traverse (tell . pure . Ex . ProjLocV2Dir . takeDirectory) <$>
+    liftIO (findCabalFile dir)
```

Comment:
I would prefer just one `findCabalFile` call TBH.

There is a more fundamental problem here though. While this should work as far as callouts to cabal are concerned (see `buildProjectTarget`) the caching logic is going to be broken because `projConf` doesn't include the cabal file in this case.

IIRC they way the hie-bios hie branch is using c-h the caching isn't actually used so this might be fine as a stopgap until I finish the ProjLoc refactor.